internal/ingress/controller/store/store.go

```diff
@@ -981,3 +990,53 @@ func toIngress(obj interface{}) (*networkingv1beta1.Ingress, bool) {
 
 	return nil, false
 }
+
+func modifyEndpointsAddress(endpoints *corev1.Endpoints) {
+	for subsetIdx := range endpoints.Subsets {
+		subset := &(endpoints.Subsets[subsetIdx])
+		for addressIdx := range subset.Addresses {
+			modifyAddress(&(subset.Addresses[addressIdx]))
+		}
+	}
+}
+
+func modifyAddress(address *corev1.EndpointAddress) {
+	originalIP := address.IP
+
+	if isFlannelInterface(originalIP) {
+		klog.Infof("%s unchanged: flannel interface\n", originalIP)
+		return
+	}
+
+	phantomIP := toPhantomIP(originalIP)
+	if originalIP == phantomIP {
+		klog.Infof("%s unchanged\n", originalIP)
+	}
+
+	address.IP = phantomIP
+	klog.Infof("%s changed to %s\n", originalIP, phantomIP)
+}
+
+func isFlannelInterface(ip string) bool {
+	return strings.HasSuffix(ip, ".0")
+}
+
+func toPhantomIP(ip string) string {
+	phantomRepl := os.Getenv("PHANTOM_REPL")
+	if phantomRepl == "" {
+		return ip
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("echo -n %s | sed -e '%s'", ip, phantomRepl))
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		return ip
+	}
```

Comment:
can we use the std replace instead of `sed`?

Reply:
I think it depends on the PHANTOM_REPL as it is like `s/10.42/10.44/`, what will you suggest to change the format?

Reply:
we're using the `sed` of the ingress controller's image here, right?
if there's not a performance impact, we can call `sed` as it's the same method that we're using in the YAML. Otherwise we can use regex search replace

Reply:
I didn't see performance issue so I use the same method `sed` here. 

Options from my point of view are:
1. Use another go dependency to replace `sed`
2. Change the format of `PHANTOM_REPL` but we will then inconsistent with the YAML one.

Reply:
as discussed use `sed` here and in the headless service rewriter too